### PR TITLE
[simd] update to 5.2.123

### LINKED ISSUE #29154: [Simd] update to 5.2.121
Library name:  Simd

New version number:  5.2.121

Fixes and enhancements:  http://ermig1979.github.io/Simd/2023.html

### PR DESCRIPTION
**Pull request by Adela0814**
Fixes: https://github.com/microsoft/vcpkg/issues/29154
Update `simd` to v5.2.123.
No feature needs to test.
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by BillyONeal — DISMISSED** [review]
I manually checked that the SHA in here is not affected by https://github.com/microsoft/vcpkg/issues/29288

It also looks good to me.

**Comment by Edi61** [comment]
New release 5.2.122 available:
- x86_windows build error is fixed
- fix-CMakeLists-install.patch is not needed anymore, because "install" target is now available in CmakeLists.

**Comment by Edi61** [comment]
Hi @Adela0814,
new release v5.2.123 incl. all fixes is now available.

File: versions/s-/simd.json [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "64cfb73836acffe8ff38e25477e0e8332707b4fc",
+      "version": "5.2.123",
+      "port-version": 0
+    },
+    {
+      "git-tree": "29cbc6d590c79f9d98ebceed9ebd6cc30dc2a1da",
+      "version": "5.2.122",
+      "port-version": 0
+    },
```
Comment (FrankXie05) [inline versions/s-/simd.json:12]:
```suggestion
```
Please remove this.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**